### ISSUE TITLE
PdfReader method `_get_outlines()` can produce outline items with incorrect "/Title"

### ISSUE THREAD

**Issue by mtd91429** [py-pdf/PyPDF2#1121]
When obtaining outlines from a PDF with bookmarks that were copied/pasted and then re-titled, some of the returned outline items point to the wrong title.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Windows-10-10.0.19044-SP0

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.5.0 (commit ed5ecd9)
Python 3.10
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from PyPDF2 import PdfReader
reader = PdfReader("mistitled_outlines_example.pdf")

def show_tree(outlines, indent=0):
    for item in outlines:
        if isinstance(item, list):
            show_tree(item, indent+4)
        else:           
            print(f'{" "*indent}{item.title}')

show_tree(reader.outlines)
```
```console
Nineteenth
    Twentieth
    Twenty-first
    Twenty-second
        Twenty-third
        Twenty-fourth
    Twenty-fifth
        Twenty-sixth
        Twenty-seventh
Nineteenth
    Twentieth
    Twenty-first
    Twenty-second
    Twenty-third
Twenty-fourth
    Twenty-fifth
    Twenty-sixth
Twenty-seventh
Nineteenth
    Twentieth
    Twenty-first
    Twenty-second
    Twenty-third
    Twenty-fourth
    Twenty-fifth
    Twenty-sixth
    Twenty-seventh
```

The expected output for this file is:
```console
First
    Second
    Third
    Fourth
        Fifth
        Sixth
    Seventh
        Eighth
        Ninth
Tenth
    Eleventh
    Twelfth
    Thirteenth
    Fourteenth
Fifteenth
    Sixteenth
    Seventeenth
Eighteenth
Nineteenth
    Twentieth
    Twenty-first
    Twenty-second
    Twenty-third
    Twenty-fourth
    Twenty-fifth
    Twenty-sixth
    Twenty-seventh
```

Here is the PDF:
[mistitled_outlines_example.pdf](https://github.com/py-pdf/PyPDF2/files/9126823/mistitled_outlines_example.pdf)

Here is a screenshot of the outline in Adobe Acrobat Reader:

![image](https://user-images.githubusercontent.com/3360192/179375962-f2dc9cb4-ae04-4034-bcbe-8b5dd1079af2.png)

**Comment by MartinThoma** [comment]
Thank you for the good example! I've added it to the unit tests. If anybody knows how to fix this, it's now easy to test it :-)

**Comment by mtd91429** [comment]
I think this has to do with how Python handles pointers and the fact that the outline objects are all recycling the same named destination. Specifically, the "First" outline entry points to the named destination "section.1", as does the "Tenth" and "Nineteenth" outline entries; the "Second", "Eleventh", and "Twentieth" point to "section.2". 

I've been stepping through the code to determine where the error is introduced, and I think it occurs at https://github.com/py-pdf/PyPDF2/blob/ae0ff49058e6c57a8edcfcd3d956665ddaa8a787/PyPDF2/_reader.py#L837

I think I have a fix and issued a pull request #1128
